docs/decisions/006-ha-distributed-query.md

```diff
@@ -0,0 +1,267 @@
+# DR-006: High Availability Distributed Query with Stateless Schedulers
+
+## Status
+
+Draft
+
+## Context
+
+Following the decisions to adopt Apache Ballista as Spice's distributed query framework (DR-004) and extend it with catalog sync, UDF sync, and security features (DR-005), Spice needs to support production-grade high availability for distributed query execution.
+
+Stock Apache Ballista uses a single-scheduler model where the scheduler is a single point of failure. If the scheduler process dies:
+
+1. All in-flight queries are lost with no recovery mechanism.
+2. Clients must manually reconnect and resubmit queries.
+3. Executor state and task progress are orphaned.
+
+For production deployments, Spice requires:
+
+* **Active-active schedulers**: Multiple scheduler nodes behind a load balancer with no leader election or failover delays.
+* **Stateless schedulers**: Any scheduler can handle any request; scheduler pods may disappear/reappear at any time.
+* **Resilient query execution**: In-flight queries should fail gracefully when a scheduler dies, with clients able to resubmit to any scheduler.
+* **Elastic scaling**: Ability to resize the cluster (add/remove executors) while queries are running.
+
+Related decisions:
+
+* [DR-004: Use Apache Ballista as Spice's distributed query framework](./004-distributed-query-framework.md)
+* [DR-005: Extend Apache Ballista with Remote Catalog, UDF Sync, and Cluster Security](./005-ballista-extensions.md)
+
+## Assumptions
+
+1. Schedulers are deployed behind a load balancer; clients connect to a single scheduler endpoint.
+2. The cluster runs on Kubernetes; executor discovery leverages K8s EndpointSlices or DNS.
```

Comment:
Based on some of the feedback we've gotten, it looks like we'll need to support cluster mode without the operator available? One option for executor discovery could be to ask the operator, but if we need this to work without the operator then we need an alternative (i.e. schedulers ask K8s directly or use DNS)